src/zap/zapinfo.cpp

```diff
@@ -440,6 +440,12 @@ void ZapInfo::CompileMethod()
 #ifdef FEATURE_READYTORUN_COMPILER
     if (IsReadyToRunCompilation())
     {
+        if (m_jitFlags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_AGGRESSIVE_OPT))
+        {
+            // Skip methods marked with MethodImplOptions.AggressiveOptimization, they will be jitted instead
+            return;
+        }
+
         // READYTORUN: FUTURE: Producedure spliting
```

Comment:
Should we do always do this (regardless of R2R or fragile ngen?) We still use fragile for corelib in 2.2 and (for now) in 3.0.

Perhaps academic for now as there aren't any `AggressiveOptimization` attributes in corelib, but somebody might add one.

Reply:
I thought that fragile ngen code would be (at the moment) as good as tier 1 JIT. Is that not the case?

Reply:
I'll change it for now, though thinking about it more, it may need to be revisited in the future for both r2r and fragile ngen. For methods that don't need any special new instructions and can be aggressively optimized, it may be worth pregenerating the code, and that would have to be determined by the JIT instead of this check.